packages/common/src/getContract.ts

```diff
@@ -69,13 +75,17 @@ export function getContract<
   TAccount,
   TPublicClient,
   TWalletClient
->): GetContractReturnType<TAbi, TPublicClient, TWalletClient, TAddress> {
-  const contract = viem_getContract<TTransport, TAddress, TAbi, TChain, TAccount, TPublicClient, TWalletClient>({
+>): GetContractReturnType<TAbi, { public: TPublicClient; wallet: TWalletClient }, TAddress> {
+  const contract = viem_getContract({
     abi,
     address,
-    publicClient,
-    walletClient,
-  }) as unknown as GetContractReturnType<Abi, PublicClient, WalletClient>;
+    client: {
+      public: publicClient,
+      wallet: walletClient,
+    },
+  }) as unknown as GetContractReturnType<TAbi, { public: TPublicClient; wallet: TWalletClient }, TAddress> & {
+    write: unknown;
```

Comment:
what does this bit do? do we need it?

Reply:
This corresponds to [this viem code](https://github.com/wevm/viem/blob/viem%402.7.12/src/actions/getContract.ts#L513-L524). The actual properties attached to the contract object depend on the clients and the abi, and since we don't know the actual abi here, we need to specify this to have the write property.
There might be a better way to handle this?

Reply:
While calling this MUD's getContract, it is typed the same as Viem's since it uses the same arguments and return type.